homeassistant/components/zeroconf/__init__.py

```diff
@@ -315,7 +296,7 @@ def service_update(zeroconf, service_type, name, state_change):
                 ):
                     continue
 
-            hass.add_job(
+            hass.async_create_task(
```

Comment:
`async_init` returns `Coroutine` while `add_job` expects `Callable`. I'm not sure if it's ok to call `async_create_task` from synchronous context.

Reply:
No, we can only use the async api inside async context. From sync context we need to use the sync api.

The core methods are not typed according to their features, fully.

Reply:
Thanks! Let's revert it then.